sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -788,7 +788,7 @@ def test_encoder_dtypes_pandas():
     assert_array_equal(enc.transform(X).toarray(), exp)
 
     X = pd.DataFrame({"A": [1, 2], "B": ["a", "b"], "C": [3.0, 4.0]})
-    X_type = [X["A"].dtype, X["B"].dtype, X["C"].dtype]
+    X_type = ["int64", "object", "float64"]
```

Comment:
Keeping the name X_type is misleading as now the dtype of column "B" is a string dtype.
```suggestion
    expected_cat_type = ["int64", "object", "float64"]
```